dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotReactiveElasticsearchClient.java

```diff
@@ -0,0 +1,874 @@
+                .map(MultiGetResponse::getResponses) //
+                .flatMap(Flux::fromArray) //
+                .filter(it -> !it.isFailed() && it.getResponse().isExists()) //
+                .map(it -> DotReactiveElasticsearchClient.getResponseToGetResult(it.getResponse()));
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#exists(org.springframework.http.HttpHeaders, org.elasticsearch.action.get.GetRequest)
+     */
+    @Override
+    public Mono<Boolean> exists(HttpHeaders headers, GetRequest getRequest) {
+
+        return sendRequest(getRequest, DotReactiveElasticsearchClient.RequestCreator.exists(), RawActionResponse.class, headers) //
+                .map(response -> response.statusCode().is2xxSuccessful()) //
+                .next();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#ping(org.springframework.http.HttpHeaders, org.elasticsearch.action.index.IndexRequest)
+     */
+    @Override
+    public Mono<IndexResponse> index(HttpHeaders headers, IndexRequest indexRequest) {
+        return sendRequest(indexRequest, DotReactiveElasticsearchClient.RequestCreator.index(), IndexResponse.class, headers).publishNext();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#indices()
+     */
+    @Override
+    public Indices indices() {
+        return this;
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#ping(org.springframework.http.HttpHeaders, org.elasticsearch.action.update.UpdateRequest)
+     */
+    @Override
+    public Mono<UpdateResponse> update(HttpHeaders headers, UpdateRequest updateRequest) {
+        return sendRequest(updateRequest, DotReactiveElasticsearchClient.RequestCreator.update(), UpdateResponse.class, headers).publishNext();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#ping(org.springframework.http.HttpHeaders, org.elasticsearch.action.delete.DeleteRequest)
+     */
+    @Override
+    public Mono<DeleteResponse> delete(HttpHeaders headers, DeleteRequest deleteRequest) {
+
+        return sendRequest(deleteRequest, DotReactiveElasticsearchClient.RequestCreator.delete(), DeleteResponse.class, headers) //
+                .publishNext();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#count(org.springframework.http.HttpHeaders, org.elasticsearch.action.search.SearchRequest)
+     */
+    @Override
+    public Mono<Long> count(HttpHeaders headers, CountRequest countRequest) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'headers' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5113278)